src/Features/Core/Portable/Completion/Providers/ExportCompletionProviderAttribute.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Completion;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    [MetadataAttribute]
+    [AttributeUsage(AttributeTargets.Class)]
+    internal class ExportCompletionProviderAttribute : ExportAttribute
+    {
+        public string Name { get; }
+        public string Language { get; }
+        public string Role { get; set; }
```

Comment:
I don't know.  I was trying to work in the ones we have defined, and I believe they are only one role each.